Explorer/Assets/DCL/SDKComponents/Tween/Components/CustomTweener/CustomTweenerImpl.cs

```diff
@@ -0,0 +1,91 @@
+﻿using CrdtEcsBridge.Components.Conversion;
+using DCL.ECSComponents;
+using DG.Tweening;
+using DG.Tweening.Core;
+using DG.Tweening.CustomPlugins;
+using DG.Tweening.Plugins.Options;
+using UnityEngine;
+
+namespace DCL.SDKComponents.Tween.Components
+{
+    public abstract class Vector3CustomTweener : CustomTweener<Vector3, VectorOptions>
+    {
+
+        protected sealed override TweenerCore<Vector3, Vector3, VectorOptions> CreateTweener(Vector3 start, Vector3 end, float duration)
+        {
+            CurrentValue = start;
+            return DOTween.To(() => CurrentValue, x => CurrentValue = x, end, duration);
+        }
+
+        public abstract override TweenResult GetResult();
+    }
+
+    public class PositionTweener : Vector3CustomTweener
+    {
+        public override TweenResult GetResult()
+        {
+            return new TweenResult
+            {
+                Position = CurrentValue, Rotation = StartTransform.localRotation, Scale = StartTransform.localScale
+            };
+        }
+
+        public override void Initialize(PBTween pbTween, Transform startTransform, float durationInSeconds)
```

Comment:
In every implementation you have a common bit of the Initialization logic. The only difference is how you get `start` and `end` values. Thus I propose you to move all repetitive logic to the base generic class and let others override variable behavior only.

P.S. there is an indicator for the wrong polymorphic behavior distribution: `CreateTweener` method is not used from the base class (as you jump to it directly from the derived classes only)